.github/workflows/ci.yml

```diff
@@ -47,31 +46,62 @@ jobs:
             coverage: true
             run_tests: true
             enabled: ${{ github.event_name != 'pull_request' }}
+            timeout: 35
           - name: linux-acr-clang-build
             os: ubuntu-latest
             build_system: acr
             compiler: clang
             enabled: ${{ github.event_name == 'pull_request' }}
+            timeout: 35
           - name: linux-meson-gcc-build
             os: ubuntu-latest
             build_system: meson
             compiler: gcc
             meson_options: -Db_coverage=true
             coverage: true
             enabled: ${{ github.event_name == 'pull_request' }}
+            timeout: 35
           - name: linux-meson-gcc-newshell-tests
             os: ubuntu-latest
             build_system: meson
             compiler: gcc
             newshell: true
             run_tests: true
             enabled: true
+            timeout: 35
           - name: macos-clang-tests
             os: macos-latest
             build_system: acr
             compiler: clang
             run_tests: true
             enabled: true
+            timeout: 35
+          - name: linux-gcc-tests-asan
+            os: ubuntu-latest
+            build_system: meson
+            compiler: gcc
+            cflags: '-DR2_ASSERT_STDOUT=1'
+            meson_options: -Db_sanitize=address,undefined
+            asan_options: 'detect_leaks=0,detect_odr_violation=0'
+            run_tests: true
+            enabled: ${{ github.event_name != 'pull_request' || contains(github.head_ref, 'asan') }}
```

Comment:
we should probably disable this for now on releases, because ASAN build has been ignored for so long that there are a lot of issues.